services/reporting-service/src/services/reports.ts

```
@@ -0,0 +1,342 @@
+  const groups = new Map();
+
+  const getGroupKey = (tx: TransactionRow, scopeId: string) => {
+    const keys: string[] = [];
+
+    groupBy.split(",").forEach((key) => {
+      switch (key.trim()) {
+        case "day":
+          keys.push(tx.initiatedAt.toISOString().split("T")[0]);
+          break;
+        case "week":
+          const dateCopy = new Date(tx.initiatedAt);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)